frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/material-feature-manager.ts

```diff
@@ -61,10 +83,39 @@ export class MaterialFeatureManager extends MoveableFeatureManager<Material> {
             },
             new PointGeometryHelper()
         );
-        this.layer.setStyle((feature, resolution) => [
-            this.nameStyleHelper.getStyle(feature as Feature, resolution),
-            this.imageStyleHelper.getStyle(feature as Feature, resolution),
-        ]);
+        this.layer.setStyle((feature, resolution) => {
+            const styles = [
+                this.nameStyleHelper.getStyle(feature as Feature, resolution),
+                this.imageStyleHelper.getStyle(feature as Feature, resolution),
+            ];
+            if (
+                this.popupService.currentPopup?.markedForTrainerUUIDs.includes(
+                    feature.getId() as UUID
+                ) &&
+                selectStateSnapshot(selectCurrentRole, this.store) === 'trainer'
```

Comment:
Since this complex conditions occur multiple times, what do you think about adding a helper `super.markedForCurrentRole()` to reduce duplicate code?

Reply:
Done